src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+						{
+							personalNumber: documentFields.document_number,
+							expiryDate: documentFields.expiration_date,
+							issueDate: documentFields.date_of_issue,
+							issuedBy: documentFields.issuing_authority,
+							fullAddress: documentFields.formatted_address,
+						},
+					];
+					break;
+				case "RESIDENCE_PERMIT": //TBC
+					credentialSubject.residencePermit = [
+						{
+							documentNumber: documentFields.document_number,
+							expiryDate: documentFields.expiration_date,
+							issueDate: documentFields.date_of_issue,
+							issuingCountry: documentFields.place_of_issue,
+						},
+					];
+					break;
+				default:
+					throw new AppError(HttpCodesEnum.SERVER_ERROR, "Invalid documentType provided", {
+						documentType, issuingCountry,
+					});
+			}
+		} else {
+			switch (documentType) {
+				case "PASSPORT":
+					credentialSubject.passport = [
+						{
+							documentNumber: documentFields.document_number,
+							expiryDate: documentFields.expiration_date,
+							icaoIssuerCode: documentFields.issuing_country,
+						},
+					];
+					break;
+				case "DRIVING_LICENCE": //TBC
+					credentialSubject.drivingPermit = [
+						{
+							personalNumber: documentFields.document_number,
+							expiryDate: documentFields.expiration_date,
+							issueDate: documentFields.date_of_issue,
+							issuedBy: documentFields.issuing_country, //TBC - May need to be mapped
+						},
+					];
+					break;
+				case "NATIONAL_ID": //TBC
+				credentialSubject.nationalId = [
+					{
+						personalNumber: documentFields.document_number,
+						expiryDate: documentFields.expiration_date,
+						issueDate: documentFields.date_of_issue,
+						issuedBy: documentFields.issuing_country, //TBC - May need to be mapped
+					},
+				];
+				break;
+				default:
+					throw new AppError(HttpCodesEnum.SERVER_ERROR, "Invalid documentType provided", {
+						documentType, issuingCountry,
+					});
+			}
+		}
+
+		
+	
+  	return credentialSubject;
+  }
+	
+
+  getVerifiedCredentialInformation(this: any, 
+  	yotiSessionId: string,
+  	completedYotiSessionPayload: any,
+  	documentFields: any,
+  ): {
+  		credentialSubject: VerifiedCredentialSubject;
+  		evidence: VerifiedCredentialEvidence;
+  	} {
+  	const { id_documents: idDocuments } = completedYotiSessionPayload.resources;
+  	const { checks } = completedYotiSessionPayload;
+  	const documentType = idDocuments[0].document_type;
+		const yotiCountryCode = idDocuments[0].issuing_country;
+	
+  	const findCheck = (type: string) =>
+  		checks.find((checkCompleted: { type: string }) => checkCompleted.type === type);
+	
+  	const getCheckObject = (check: { state: any; report: { recommendation: any; breakdown: any } }) => ({
+  		object: check,
+  		state: check.state,
+  		recommendation: check.report.recommendation,
+  		breakdown: check.report.breakdown,
+  	});
+	
+		//IBV_VISUAL_REVIEW_CHECK && DOCUMENT_SCHEME_VALIDITY && PROFILE_DOCUMENT_MATCH Currently not being consumed
```

Comment:
what's the comment here for - is this a problem or a TODO?

Reply:
It's a TODO for IF and When the last 3 scenarios in [this doc](https://govukverify.atlassian.net/wiki/spaces/FTFCRI/pages/3557818744/Yoti+Journey+4+Scenarios) are decided on